packages/core/src/server/compilerDevMiddleware.ts

```diff
@@ -122,14 +123,16 @@ export class CompilerDevMiddleware {
       etag: 'weak',
     });
 
+    const assetPrefixes = publicPaths.map(pathParse);
+
     const warp = async (
       req: IncomingMessage,
       res: ServerResponse,
       next: NextFunction,
     ) => {
       const { url } = req;
       const assetPrefix =
-        url && publicPaths.find((prefix) => url.startsWith(prefix));
+        url && assetPrefixes.find((prefix) => url.startsWith(prefix));
```

Comment:
Maybe several assetPrefixes can match the same url?

- assetPrefixA: `/foo/bar`
- assetPrefixB: `/foo`

Reply:
It's possible. Users are currently required to avoid this situation as it is difficult to handle on the server side, especially when the filenames are the same.

Reply:
get